src/machine.rs

```diff
@@ -954,7 +954,7 @@ impl<'tcx> Machine<'tcx> for MiriMachine<'tcx> {
             // foreign function
             // Any needed call to `goto_block` will be performed by `emulate_foreign_item`.
             let args = ecx.copy_fn_args(args); // FIXME: Should `InPlace` arguments be reset to uninit?
-            let link_name = ecx.item_link_name(instance.def_id());
+            let link_name = Symbol::intern(ecx.tcx.symbol_name(instance).name);
             return ecx.emulate_foreign_item(link_name, abi, &args, dest, ret, unwind);
```

Comment:
emulate_foreign_item anyway makes this an `&str`... so the interning seems unnecessary here? It's probably overall easier to pass around `&str`.

Reply:
`symbol_name` returns a `SymbolName<'tcx>` which is a wrapper around `&'tcx str`. The extra lifetime is giving me quite a bit of trouble. I could pass around String instead, but that is likely less efficient.

Reply:
I thought it'd be trivial to just make this `&str` everywhere (so `emulate_foreign_item` would not even know that the lifetime is `tcx`)... but I guess it is not that simple, then?

Reply:
There are a couple of types without a 'tcx lifetime, some functions return a symbol name as `String` and I also saw a bunch of borrowck errors for whatever reason.

Reply:
:shrug: okay not sure what happens and I don't have the time to take a look myself right now